File: client_sock.py
import socket

def start_client():
    # Create a socket object
    client_socket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)

    # Connect to the server
    server_address = ('localhost', 65430)
    client_socket.connect(server_address)
    print("Connected to server:", server_address)

    # Send data to the server
    message = "Hello from the client!"
    client_socket.sendall(message.encode())

    # Receive a response from the server
    response = client_socket.recv(1024).decode()
    print("Received response:", response)

    # Close the connection
    client_socket.close()

if __name__ == '__main__':
    start_client()

File: client_sock.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <unistd.h>
#include <sys/socket.h>
#include <netinet/in.h>
#include <arpa/inet.h>

#define PORT 65430
#define BUFFER_SIZE 1024

int main() {
    int sock = 0, valread;
    struct sockaddr_in serv_addr;
    char *message = "Hello from client!";
    char buffer[BUFFER_SIZE] = {0};

    // Create a socket
    if ((sock = socket(AF_INET, SOCK_STREAM, 0)) < 0) {
        printf("\n Socket creation error \n");
        return -1;
    }

    serv_addr.sin_family = AF_INET;
    serv_addr.sin_port = htons(PORT);

    // Convert IPv4 and IPv6 addresses from text to binary form
    if (inet_pton(AF_INET, "127.0.0.1", &serv_addr.sin_addr) <= 0) {
        printf("\nInvalid address/ Address not supported \n");
        return -1;
    }

    // Connect to the server
    if (connect(sock, (struct sockaddr *)&serv_addr, sizeof(serv_addr)) < 0) {
        printf("\nConnection Failed \n");
        return -1;
    }

    // Send a message to the server
    send(sock, message, strlen(message), 0);
    printf("Message sent to the server\n");

    // Read the server response
    valread = read(sock, buffer, BUFFER_SIZE);
    printf("Server response: %s\n", buffer);

    // Close the socket
    close(sock);

    return 0;
}
